packages/common-all/src/store/IConfigStore.ts

```diff
@@ -0,0 +1,59 @@
+import { ResultAsync } from "neverthrow";
+import { IDendronError } from "../error";
+import { DendronConfig, DendronConfigValue } from "../types";
+import { DeepPartial } from "../utils";
+
+export type ConfigValue = string | number | object;
+
+export type ConfigReadMode = "override" | "default";
+export type ConfigReadOpts = {
+  mode: ConfigReadMode;
+  useCache?: boolean;
+};
+
+export type ConfigGetOpts = ConfigReadOpts;
```

Comment:
do we need a separate type if the underlying struct is the same?

Reply:
not really, preemptively distinguished them thinking it may have more props. will be cleaned up